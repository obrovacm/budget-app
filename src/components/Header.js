import React from "react";
import { NavLink } from "react-router-dom";

const Header = () => (
  <header>
    <h1>Budget app</h1>
    <NavLink
      to="/"
      exact={true}
      activeClassName="is-active"
      className="nav-link"
    >
      Dashboard
    </NavLink>
    <NavLink to="/add-expense" activeClassName="is-active" className="nav-link">
      Add Expense
    </NavLink>
  </header>
);

export default Header;
