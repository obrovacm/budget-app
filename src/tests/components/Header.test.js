import React from "react";
import { shallow } from "enzyme";

//// provides html without superfluous info
// import toJSON from "enzyme-to-json";
//// but import is not needed when you mention it in jes.config.json
import Header from "../../components/Header";

test("should render Header correctly", () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
});

// expect(toJSON(wrapper)).toMatchSnapshot();

// expect(wrapper.find("h1").text()).toBe("Budget app");

//// since we use enzyme 'ReactShallowRenderer' is not needed
// const renderer = new ReactShallowRenderer();
// renderer.render(<Header />);
// expect(renderer.getRenderOutput()).toMatchSnapshot();
