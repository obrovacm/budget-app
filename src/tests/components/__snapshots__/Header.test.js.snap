// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Budget app
  </h1>
  <ForwardRef
    activeClassName="is-active"
    className="nav-link"
    exact={true}
    to="/"
  >
    Dashboard
  </ForwardRef>
  <ForwardRef
    activeClassName="is-active"
    className="nav-link"
    to="/add-expense"
  >
    Add Expense
  </ForwardRef>
</header>
`;
